### PR TITLE
Globalize getColorForState function

### PR DESCRIPTION
**Pull request by SourceDoctor**
globalizing getColorForState function

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
I don't use any of these pieces of code. Let's get some testers for this before merge.

**Comment by SourceDoctor** [comment]
@PipoCanaja 
maybe splitting up this PR so single parts could be merged , instead of waiting for all is retested?

File: LibreNMS/Alert/Transport/Hipchat.php [hunk 1]
```diff
@@ -62,20 +62,7 @@ public function contactHipchat($obj, $option)
...
-            $color = "yellow";
-        } else {
-            if (empty($option["color"]) || $option["color"] == 'u') {
-                $color = 'red';
-            } else {
-                $color = $option["color"];
-            }
-        }
-
+        $color = self::getColorForState($obj['state']);
```
Comment (Jellyfrog) [inline LibreNMS/Alert/Transport/Hipchat.php]: Does hipchat support hexcolors?

Comment (SourceDoctor) [reply LibreNMS/Alert/Transport/Hipchat.php]: As i see in documentation, it should.
Alternative an extension of `getColorForState` function to return named colors would be an option.
As PipoCanaja said, someone needs to test it. I also don't use Hipchat.
Slack i can test myself :-)

Comment (SourceDoctor) [reply LibreNMS/Alert/Transport/Hipchat.php]: ok, my fault. i looked on the false places.
Only a enum of colors is allowed.
I reverted this change.

**Comment by SourceDoctor** [comment]
@PipoCanaja @Jellyfrog 
i reduced PR to refactor only, so it can be merged more easily.
Code changes in Transports will be PR per Transport

**Comment by PipoCanaja** [comment]
OK. You'll have to rebase/merge master in the other PRs.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-59-release-changelog-december-2019/10558/1

**Pull request merged**